tests/main/views/test_users.py

```diff
@@ -1411,6 +1411,22 @@ def _assert_things_about_export_response(self, row, parameters=None):
             'application_status': 'no_application',
             'declaration_status': 'unstarted',
             'framework_agreement': False,
+            'supplier_name': "Supplier 1",
+            'supplier_organisation_size': "small",
+            'duns_number': "100000001",
+            'registered_name': 'Registered Supplier Name 1',
+            'companies_house_number': None,
+            'contact_name': 'Contact for Supplier 1',
+            'contact_email': '1@contact.com',
+            'contact_phone_number': None,
+            'address_first_line': '7 Gem Lane',
+            'address_city': 'Cantelot',
+            'address_postcode': 'CN1A 1AA',
+            'address_country': 'country:GB',
+            'published_services_count_on_digital-outcomes_lot': 0,
+            'published_services_count_on_digital-specialists_lot': 0,
+            'published_services_count_on_user-research-studios_lot': 0,
+            'published_services_count_on_user-research-participants_lot': 0,
```

Comment:
Why have we decided to flatten these counts into the top level object with the lot names string-formatted in? Why is it not

```
...
"published_services_count": {
    "digital-outcomes": 0,
    "digital-specialists": 0,
    "user-research-studios": 0,
    "user-research-participants": 0,
}
...
```

Reply:
Yes, nested data like in your comment looks neater indeed. 

Things that would need to be explored are:
- how it would work when CSV is created in this view
- how framework managers at CCS are using this data

Reply:
Mhm, it's the script/frontend's business whether it's turning it into a csv or not. If we're wed to it being output in csv-like format we should just output a csv directly.

Reply:
I think the problem here is that we're trying to get a lot of (mostly duplicated!) data from one API call. The resulting CSV will display everything as per-user, but I think it'll probably be easier to split the per-supplier info (including service count per lot) out into a new API view, and then join that to each supplier user in the Admin FE (using @risicle's suggested code above). It does mean 2 API calls so it adds that overhead, but they should be smaller in size?

Reply:
Um, yea? I don't really know the "points of use" of this endpoint in question so :man_shrugging: ... I guess that would also lead to a way of providing the ability to request the export in "cheap mode" without any aggregate data.

Offtopic... though it would sully the "JSON purity" of our wonderful API I think at some point we should really entertain producing csv directly from the API - csv can be produced directly from database output with near-zero memory overhead, the act of making this json uses potentially numerous MBs on the API instances simply in the dict-representation of the output, not to mention any intermediary dicts & list copies we use during preparation. Then we spend approximately that amount of memory *again* decoding the bytes back into dicts/lists on the frontend instances, manipulating them.... you get my drift. All seems like a bit of a silly dance when we start to run into problems with the size of our exports...